src/openms/include/OpenMS/ANALYSIS/OPENSWATH/PeakIntegrator.h

```diff
@@ -567,6 +630,67 @@ namespace OpenMS
       const bool is_left_half
     ) const;
 
+    /**
+      @brief Given a peak, extract a training set to be used with the gradient descent algorithm
+
+      The algorithm tries to select only those points that can help in finding the optimal
+      parameters with gradient descent. The decision of which points to skip is based on the
+      derivatives between consecutive points.
+
+      It first selects all those points whose intensity is below a certain value (`intensity_threshold`).
+      Then, the derivatives of all the remaining points are computed. Based on the results,
+      the algorithm selects those points that present a high enough derivative.
+      Once a low value is found, the algorithm stops taking points from that side.
+      It then repeats the same procedure on the other side of the peak.
+      If points are actually skipped, these will be found in the middle part of the peak,
+      possibly being the saturated region.
```

Comment:
Replace lines 645-646 with the following:  "The goal is to limit the inclusion of saturated or spurious points near the peak apex during training."